test/controllers/unattended_controller_test.rb

```diff
@@ -373,6 +373,16 @@ class UnattendedControllerTest < ActionController::TestCase
       refute nic.build
     end
 
+    test "should accept built notifications over ipv6 and store the address" do
+      nic6 = FactoryBot.build(:nic_managed, :with_ipv6)
+      Setting[:update_ip_from_built_request] = true
+      @request.env["REMOTE_ADDR"] = nic6.ip6
```

Comment:
Isn't it easier to just use some specific IP instead of using the NIC factory?

Reply:
I didn't want to copy the IP generation code, so having FactoryBot actually generating one was an easy option for me.

Reply:
You don't need a random IP, a static IP from the `2001:db8::/32` range is IMHO better. That way you can assert for it. Your current code could have a bug where both values are set to `nil` and you wouldn't notice it.